forui/lib/src/widgets/line_calendar/line_calendar.dart

```diff
@@ -0,0 +1,346 @@
+        assert(
+          start == null || end == null || start.toLocalDate() < end.toLocalDate(),
+          'end date must be greater than start date',
+        ),
+        today = today ?? DateTime.now();
+
+  @override
+  State<FLineCalendar> createState() => _FLineCalendarState();
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      ..add(DiagnosticsProperty('style', style))
+      ..add(DiagnosticsProperty('controller', controller))
+      ..add(DiagnosticsProperty('start', start))
+      ..add(DiagnosticsProperty('end', end))
+      ..add(DiagnosticsProperty('today', today));
+  }
+}
+
+class _FLineCalendarState extends State<FLineCalendar> {
+  late ScrollController _controller;
+  late double _size;
+  late FLineCalendarStyle _style;
+
+  @override
+  void didChangeDependencies() {
+    _style = widget.style ?? FTheme.of(context).lineCalendarStyle;
+    _size = _calculateSize(context, _style);
+
+    final value = widget.controller.value?.toLocalDate() ?? widget.today.toLocalDate();
+    final offset = (value.difference(widget.start.toLocalDate()).inDays - 2) * _size + _style.itemPadding;
+    _controller = ScrollController(initialScrollOffset: offset);
+
+    super.didChangeDependencies();
+  }
+
+  // TODO: calculate width of items based on the text font size.
+  double _calculateSize(BuildContext context, FLineCalendarStyle style) {
+    final textScalor = MediaQuery.textScalerOf(context);
+    final dateTextSize = textScalor.scale(style.content.unselectedDateTextStyle.fontSize!);
+    final dayTextSize = textScalor.scale(style.content.unselectedDayTextStyle.fontSize!);
+    return dateTextSize + dayTextSize + _textSpacing + (style.content.verticalPadding * 2);
+  }
+
+  @override
+  Widget build(BuildContext context) => SizedBox(
+        height: _size,
+        child: Focus(
+          autofocus: true,
+          onKeyEvent: (node, event) {
+            if (event is KeyDownEvent && event.logicalKey == LogicalKeyboardKey.arrowRight) {
+              widget.controller.value = widget.controller.value?.add(const Duration(days: 1));
+              _controller.animateTo(
+                _controller.offset + _size,
+                duration: const Duration(milliseconds: 100),
+                curve: Curves.easeInOut,
+              );
+              return KeyEventResult.handled;
+            } else if (event is KeyDownEvent && event.logicalKey == LogicalKeyboardKey.arrowLeft) {
+              widget.controller.value = widget.controller.value?.subtract(const Duration(days: 1));
+              _controller.animateTo(
+                _controller.offset - _size,
+                duration: const Duration(milliseconds: 100),
+                curve: Curves.easeInOut,
+              );
+              return KeyEventResult.handled;
+            }
+            return KeyEventResult.ignored;
+          },
```

Comment:
I think this implementation will be better if the [`Shortcuts`](https://api.flutter.dev/flutter/widgets/Shortcuts-class.html) widget is used instead. See https://docs.flutter.dev/ui/interactivity/actions-and-shortcuts for more information.